### DISCUSSION TITLE
[Feature] Accepting Affinity photo files

### DISCUSSION THREAD

**Discussion by coissac** [immich-app/immich#4022] [Feature Request]
### The feature

I'm here speaking about affinity photo, but I think it could also apply to other photo edition software formats.
Adding such capacity would allow users to use immich to store in a same place everything related to their numerical photo. I'm testing immich for more than a month. I have imported every numeric photo I have since 2001, more than 80,000 files, and it is my first opportunity since ever to dive into them with pleasure. If I can add my edition files (currently done using affinity photo) it would be so great.

It should be effortless to implement. The file includes a PNG thumbnail that can easily extracted using a small JavaScript code : 

```js
const fs = require("fs");

// png spec: https://www.w3.org/TR/PNG/
const PNG_SIG = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
const IEND_SIG = Buffer.from([73, 69, 78, 68]);

function extractPngBlob(buf) {
  const start = buf.indexOf(PNG_SIG);
  const end = buf.indexOf(IEND_SIG, start) + IEND_SIG.length * 2; // IEND + CRC
  return buf.subarray(start, end);
}

function extractThumbnail(input, output) {
  const buf = fs.readFileSync(input);
  const pngBlob = extractPngBlob(buf);
  fs.writeFileSync(output, pngBlob);
}

extractThumbnail(process.argv[2], process.argv[3] || "output.png");
```
The code is from the following url: https://uechi.io/blog/affinity-thumbnail/

I tested it, and it produces, as expected, a correct PNG file.

I added a stupid zipped affinity photo test file.

[test_file.afphoto.zip](https://github.com/immich-app/immich/files/12556769/test_file.afphoto.zip)

All the best.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by michaelpetik** [comment]
Given Affinity's switch to a freemium model, I'd expect .afphoto to become an increasingly common file type and I'd love to see support for it in immich.
